app/src/main/java/com/dragic/gamehunter/view/home/HomeScreen.kt

```diff
@@ -51,17 +55,33 @@ fun HomeScreen(
                 .padding(paddingValues)
         ) {
             Column {
-                Text(
+                Row(
                     modifier = Modifier
+                        .fillMaxWidth()
                         .padding(
                             top = dimensionResource(id = R.dimen.home_text_padding),
+                            bottom = dimensionResource(id = R.dimen.text_padding_medium),
                             start = dimensionResource(id = R.dimen.home_top_deals_label_horizontal_padding),
                             end = dimensionResource(id = R.dimen.home_top_deals_label_horizontal_padding),
-                            bottom = dimensionResource(id = R.dimen.text_padding_medium),
                         ),
-                    text = stringResource(id = R.string.top_deals),
-                    style = MaterialTheme.typography.labelLarge
-                )
+                    horizontalArrangement = Arrangement.SpaceBetween,
+                    verticalAlignment = Alignment.CenterVertically,
+                ) {
+                    Text(
+                        text = stringResource(id = R.string.top_deals),
+                        style = MaterialTheme.typography.labelLarge
+                    )
+                    Row(verticalAlignment = Alignment.CenterVertically) {
+                        Text(text = "Dark mode ")
+                        Switch(
+                            checked = switchState,
+                            onCheckedChange = {
+                                switchState = it
+                                onThemeChange(theme)
```

Comment:
If for some reason the theme does not change after you call `onThemeChange`, the switch will be toggled, but the theme will stay the same. This is because you've implemented `switchState` separately from the `theme`. Instead, you should do something like this: 

```kt
Switch(
  checked = theme == AppTheme.DARK,
  onCheckedChange = onThemeChange
)
```

That way the switch's state will always reflect the currently applied theme.